x-pack/plugins/lists/public/exceptions/hooks/use_exception_list.test.ts

```diff
@@ -68,203 +68,398 @@ describe('useExceptionList', () => {
-    const spyOnfetchExceptionListItemsByListId = jest.spyOn(api, 'fetchExceptionListItemsByListId');
+  test('fetches only detection list items if "showDetectionsListsOnly" is true', async () => {
+    const spyOnfetchExceptionListsItemsByListIds = jest.spyOn(
+      api,
+      'fetchExceptionListsItemsByListIds'
+    );
+
+    await act(async () => {
+      const onSuccessMock = jest.fn();
+      const { waitForNextUpdate } = renderHook<UseExceptionListProps, ReturnExceptionListAndItems>(
+        () =>
+          useExceptionList({
+            filterOptions: [],
+            http: mockKibanaHttpService,
+            lists: [
+              { id: 'myListId', listId: 'list_id', namespaceType: 'single', type: 'detection' },
+              {
+                id: 'myListIdEndpoint',
+                listId: 'list_id_endpoint',
+                namespaceType: 'agnostic',
+                type: 'endpoint',
+              },
+            ],
+            matchFilters: false,
+            onError: onErrorMock,
+            onSuccess: onSuccessMock,
+            pagination: {
+              page: 1,
+              perPage: 20,
+              total: 0,
+            },
+            showDetectionsListsOnly: true,
+            showEndpointListsOnly: false,
+          })
+      );
+      await waitForNextUpdate();
+      await waitForNextUpdate();
+
+      expect(spyOnfetchExceptionListsItemsByListIds).toHaveBeenCalledWith({
+        filterOptions: [],
+        http: mockKibanaHttpService,
+        listIds: ['list_id'],
+        namespaceTypes: ['single'],
+        pagination: { page: 1, perPage: 20 },
+        signal: new AbortController().signal,
+      });
+    });
+  });
+
+  test('fetches only detection list items if "showEndpointListsOnly" is true', async () => {
+    const spyOnfetchExceptionListsItemsByListIds = jest.spyOn(
+      api,
+      'fetchExceptionListsItemsByListIds'
+    );
+
+    await act(async () => {
+      const onSuccessMock = jest.fn();
+      const { waitForNextUpdate } = renderHook<UseExceptionListProps, ReturnExceptionListAndItems>(
+        () =>
+          useExceptionList({
+            filterOptions: [],
+            http: mockKibanaHttpService,
+            lists: [
+              { id: 'myListId', listId: 'list_id', namespaceType: 'single', type: 'detection' },
+              {
+                id: 'myListIdEndpoint',
+                listId: 'list_id_endpoint',
+                namespaceType: 'agnostic',
+                type: 'endpoint',
+              },
+            ],
+            matchFilters: false,
+            onError: onErrorMock,
+            onSuccess: onSuccessMock,
+            pagination: {
+              page: 1,
+              perPage: 20,
+              total: 0,
+            },
+            showDetectionsListsOnly: false,
+            showEndpointListsOnly: true,
+          })
+      );
+      await waitForNextUpdate();
+      await waitForNextUpdate();
```

Comment:
Interesting there has to be two updates before the expect. I trust it, but that is interesting to me. Is there a comment of why two are needed we can add here? Otherwise I always will be like 🤔 when I read this or work with it later.

Optional change of a comment. Not required.

Reply:
So did a bit of digging. The first one is initialization. I added a note to them all.